src/core/containers/ServerHtml.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component, PropTypes } from 'react';
+import ReactDOM from 'react-dom/server';
+import serialize from 'serialize-javascript';
+import Helmet from 'react-helmet';
+
+
+export default class ServerHtml extends Component {
+
+  static propTypes = {
+    appName: PropTypes.string.isRequired,
+    assets: PropTypes.object.isRequired,
+    component: PropTypes.object.isRequired,
+    env: PropTypes.string.isRequired,
+    sriData: PropTypes.object,
+    store: PropTypes.object.isRequired,
+  };
+
+  getStatic({filePath, type} = {}) {
+    const { env, sriData, appName } = this.props;
+    const leafName = filePath.split('/').pop();
+    const isDeployed = ['stage', 'dev', 'production'].indexOf(env) > -1;
+    let sriProps = {};
+    // Only output files for the current app.
+    if (leafName.startsWith(appName)) {
+      if (isDeployed) {
+        sriProps = {
+          integrity: sriData[leafName],
+          crossOrigin: 'anonymous',
+        };
+        if (!sriProps.integrity) {
+          throw new Error(`SRI Data is missing for ${leafName}`);
+        }
+      }
+      switch (type) {
+        case 'css':
+          return (<link href={filePath} {...sriProps}
+                       rel="stylesheet" type="text/css" />);
+        case 'js':
+          return <script src={filePath} {...sriProps}></script>;
+        default:
+          throw new Error('Unknown static type');
+      }
+    } else {
+      return null;
+    }
+  }
+
+  getStyle() {
+    const { assets } = this.props;
+    return Object.keys(assets.styles).map((style) =>
+      this.getStatic({filePath: assets.styles[style], type: 'css'}));
+  }
+
+  getScript() {
+    const { assets } = this.props;
+    return Object.keys(assets.javascript).map((js) =>
+      this.getStatic({filePath: assets.javascript[js], type: 'js'}));
+  }
+
+  render() {
+    const { component, store } = this.props;
+    const content = component ? ReactDOM.renderToString(component) : '';
+    const head = Helmet.rewind();
+    const htmlAttrs = head.htmlAttributes.toComponent();
+
+    return (
+      <html {...htmlAttrs}>
+        <head>
+          <meta charSet="utf-8" />
```

Comment:
Yes the test caught that not working. JSX is full of those kind of weird camel attr names like `className` instead of `classname`.